src/jitlayers.cpp

```diff
@@ -588,19 +588,21 @@ CompilerResultT JuliaOJIT::CompilerT::operator()(Module &M)
     JL_TIMING(LLVM_OPT);
 
     int optlevel;
+    int optlevel_modmin;
     if (jl_generating_output()) {
         optlevel = 0;
     }
     else {
         optlevel = jl_options.opt_level;
+        optlevel_modmin = jl_options.opt_level_min;
         for (auto &F : M.functions()) {
             if (!F.getBasicBlockList().empty()) {
                 Attribute attr = F.getFnAttribute("julia-optimization-level");
                 StringRef val = attr.getValueAsString();
                 if (val != "") {
                     int ol = (int)val[0] - '0';
                     if (ol >= 0 && ol < optlevel)
-                        optlevel = ol;
+                        optlevel = std::max(ol, optlevel_modmin);
```

Comment:
Instead this should be applied once after the loop, in case it's greater than jl_options.opt_level.